### PR TITLE
NMS-13231: Backport Security Issues from Last Month

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-13231

### CODE REVIEW

File: smoke-test/src/test/java/org/opennms/smoketest/NotificationsPageIT.java [hunk 1]
```diff
@@ -29,15 +29,26 @@
...
 import org.openqa.selenium.By;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @FixMethodOrder(MethodSorters.NAME_ASCENDING)
 public class NotificationsPageIT extends OpenNMSSeleniumIT {
+    private static final Logger LOG = LoggerFactory.getLogger(NotificationsPageIT.class);
+
```
Comment (fooker) [inline smoke-test/src/test/java/org/opennms/smoketest/NotificationsPageIT.java:51]: This is unused

File: core/web-assets/src/test/javascript/ng-requisitions/controllers/ForeignSource.test.js [hunk 2]
```diff
@@ -6,6 +6,7 @@
 'use strict';
 
 const angular = require('angular-js');
+const _ = require('underscore-js');
```
Comment (fooker) [inline core/web-assets/src/test/javascript/ng-requisitions/controllers/ForeignSource.test.js:9]: This is unused

File: core/web-assets/src/test/javascript/ng-requisitions/controllers/Node.test.js [hunk 3]
```diff
@@ -8,6 +8,7 @@
 'use strict';
 
 const angular = require('angular-js');
+const _ = require('underscore-js');
```
Comment (fooker) [inline core/web-assets/src/test/javascript/ng-requisitions/controllers/Node.test.js:11]: This is unused

File: core/web-assets/src/test/javascript/ng-requisitions/controllers/Requisition.test.js [hunk 4]
```diff
@@ -8,6 +8,7 @@
 'use strict';
 
 const angular = require('angular-js');
+const _ = require('underscore-js');
```
Comment (fooker) [inline core/web-assets/src/test/javascript/ng-requisitions/controllers/Requisition.test.js:11]: This is unused

File: core/web-assets/src/test/javascript/ng-requisitions/controllers/Requisitions.test.js [hunk 5]
```diff
@@ -8,6 +8,7 @@
 'use strict';
 
 const angular = require('angular-js');
+const _ = require('underscore-js');
```
Comment (fooker) [inline core/web-assets/src/test/javascript/ng-requisitions/controllers/Requisitions.test.js:11]: This is unused

**Review by fooker — CHANGES_REQUESTED** [review]
There are some unused code fragments left over from back-porting.

File: core/web-assets/src/test/javascript/ng-requisitions/controllers/ForeignSource.test.js (hunk 2 above)
Comment (christianpape) [reply core/web-assets/src/test/javascript/ng-requisitions/controllers/ForeignSource.test.js:9]: Without this entry the tests fail.

File: smoke-test/src/test/java/org/opennms/smoketest/NotificationsPageIT.java (hunk 1 above)
Comment (christianpape) [reply smoke-test/src/test/java/org/opennms/smoketest/NotificationsPageIT.java:51]: Fixed.

File: core/web-assets/src/test/javascript/ng-requisitions/controllers/Node.test.js (hunk 3 above)
Comment (christianpape) [reply core/web-assets/src/test/javascript/ng-requisitions/controllers/Node.test.js:11]: Without this entry the tests fail.

File: core/web-assets/src/test/javascript/ng-requisitions/controllers/Requisition.test.js (hunk 4 above)
Comment (christianpape) [reply core/web-assets/src/test/javascript/ng-requisitions/controllers/Requisition.test.js:11]: Without this entry the tests fail.

File: core/web-assets/src/test/javascript/ng-requisitions/controllers/Requisitions.test.js (hunk 5 above)
Comment (christianpape) [reply core/web-assets/src/test/javascript/ng-requisitions/controllers/Requisitions.test.js:11]: Without this entry the tests fail.

**Review by RangerRick — APPROVED** [review]

**Review by fooker — APPROVED** [review]

**Pull request merged**